packages/server/src/documents/documents.controller.ts

```diff
@@ -48,20 +48,36 @@ export class DocumentsController {
       },
     );
 
+    createResult.document_status = 'Creating...';
+    createResult.nft_status = 'No NFT minted';
+
+    const created = await this.databaseService.documents.insert({
+      ...createResult,
+      ownerId: request.user._id,
+    });
     const createAttributes = unflatten(createResult.attributes);
     createResult.attributes = createAttributes;
-    // @ts-ignore
     const commitResult = await this.centrifugeService.documents.commitDocumentV2(
         request.user.account,
         // @ts-ignore
         createResult.header.document_id,
     );
+
     // @ts-ignore
-    await this.centrifugeService.pullForJobComplete(commitResult.header.job_id, request.user.account);
-    return await this.databaseService.documents.insert({
-      ...createResult,
-      ownerId: request.user._id,
-    });
+    const commit = await this.centrifugeService.pullForJobComplete(commitResult.header.job_id, request.user.account);
+    if (commit.status === 'success') {
+      return await this.databaseService.documents.updateById(created._id, {
```

Comment:
To discuss: `updateById` versions the document? A failed request will overwrite a previous valid document?

Reply:
nestDB is an append only database, so updating a document does create a new version in the DB. in this case, updateById only triggers if the job returns success, so a failed request will not update the nestDB object